### DISCUSSION TITLE
Feature Request: Gradient Buttons for Default Colors

### DISCUSSION THREAD

**Discussion by rikp777** [saadeghi/daisyui#1803] [Ideas / feature requests]
Hello,

I'm writing to propose an enhancement to the DaisyUI plugin for Tailwind CSS. Specifically, I believe it would be a valuable addition to have gradient buttons as part of the default color options.

Currently, the default color options for buttons in DaisyUI are solid, which, while functional, do not offer the same dynamic and visually engaging feel that gradient colors can provide. By integrating gradient color options for buttons, I believe that we could significantly enhance the aesthetic appeal and modernity of user interfaces built with DaisyUI, potentially increasing user interaction and satisfaction.

I have included a couple of GIFs below to demonstrate what these gradient buttons might look like:
![firefox_lVI1r4wNiC](https://github.com/saadeghi/daisyui/assets/33726176/05fdc776-35f6-4fad-9eb5-20599ffa8dbb)
![firefox_pKYbqCoSt4](https://github.com/saadeghi/daisyui/assets/33726176/35b56e48-3aaa-4626-96fa-045e1b3e7c55)
![firefox_wU7qgKoW3s](https://github.com/saadeghi/daisyui/assets/33726176/69608ae1-0930-4ff7-af33-d84d58444d60)

Used like: 
```html 
<button class="btn btn-gradient btn-error">Error Button</button>
```

**Comment by saadeghi** [comment]
That's really easy to create using Tailwind CSS gradient utility classes:
```html
<button class="btn btn-primary bg-gradient-to-l from-white/30">primary</button> 
```

More examples: https://play.tailwindcss.com/wkbaRHudO3

So adding it to daisyUI would be just combining 2 class names `bg-gradient-to-l from-white/30` to one `btn-gradient` but unfortunately it would take the design decision from you because there are many ways to create such gradient (direction, opacity, colors, etc...) and I think using Tailwind CSS classes for this makes more sense.

Let me know if you have any questions.

**Reply by rikp777** [reply]
I understand what you mean. I'm actually using the same approach myself, but I agree that it would be great if gradient buttons were standardized within DaisyUI. Outlined buttons are already a standardized option, and I think gradient buttons could be seen in the same way.

By providing a standard way of creating gradient buttons, we could reduce the amount of custom CSS required and make it easier for developers to create visually engaging interfaces. It could also promote consistency in design and improve the user experience for users who interact with interfaces built with DaisyUI.

**Reply by saadeghi** [reply]
Here's why I don't see this as a good addition:  
For UI libraries, I see a spectrum of abstraction like this:
```
        0%                                                               100%
        | -------------------------------|------------------------------- |
Minimum abstraction                                                  Maximum abstraction
```
The more we go left, there would be less rules, less design decisions and no design system tokens. Imagine writing vanilla CSS without any design token and no consistency. But you would be free to do absolutely anything with the design of each element.  
The more we go right, it would be a super strict design system and you can't [easily] customize things.  

I see libraries like Material Design or Bootstrap on the right side of spectrum. They suggest you strict rules on how a button should look like and it's super hard to customize the initial design decisions.
And you can see Tailwind CSS (Not to be confused with Tailwind UI) on the left side of spectrum. It only provides some basic design tokens for spacing, font sizes, colors, etc but has no opinions about the design decisions of a button.  

The more we go left, it allows people to create unique websites but they need to spend more time on design decisions.  
The more we go right, it allows people to make things faster using the provided design decisions but customizing the design is harder.  
That's why almost every website made with Bootstrap or Material design look exactly the same and websites made with Tailwind CSS (Not Tailwind UI) look really unique.

I'm trying to keep daisyUI somewhere in the middle. I want to provide some levels of design decisions for elements but I still want to encourage developers to use Tailwind CSS utility classes for customization.  
I don't want every website created with daisyUI to look the same. So instead of using colors like `blue`, `red` etc, for the components, I used semantic names like `primary`, `secondary`, etc to make it easy to create a new unique theme for each project.  

I know there are some components in daisyUI with less or more strictness about design decision and customizability. Some components have no `modifer` class name and some have more than 10. But overall, I think currently there's a good [balance](https://i.giphy.com/media/Ry1MOAeAYXvRVQLPw3/giphy.webp) on the amount of design decisions daisyUI provides to help developers style elements faster and the amount of decisions that should be handled using utility classes.  

For buttons, I think there's a difference between `outline` modifier class names and gradient buttons: Outline buttons are necessary for visual hierarchy sometimes. For example sometimes you want a button to be visible but you don't want it to be eye catching with a lot of contrast. I think in this case it's good to have a transparent button with border. But gradient on the other hand is a design fanciness. It's like having box shadow or glow or animations. It won't change the visual hierarchy of a solid, flat button comparing to other buttons.
